File: src/main/java/com/careeros/common/model/Result.java
package com.careeros.common.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Result<T> implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    private Integer code;

    private String msg;

    private T data;

    public static <T> Result<T> success(T data) {
        return Result.<T>builder()
                .code(200)
                .msg("success")
                .data(data)
                .build();
    }

    public static <T> Result<T> fail(Integer code, String msg) {
        return Result.<T>builder()
                .code(code)
                .msg(msg)
                .data(null)
                .build();
    }
}

File: Hai-agent/app/common/schemas/result.py
from typing import Generic, TypeVar

from pydantic import BaseModel


T = TypeVar("T")


class Result(BaseModel, Generic[T]):
    """统一接口响应，所有接口均返回 code、msg 和 data。"""

    code: int
    msg: str
    data: T | None = None

    @classmethod
    def success(cls, data: T | None = None, msg: str = "success") -> "Result[T]":
        """创建统一成功响应。"""
        return cls(code=0, msg=msg, data=data)

    @classmethod
    def fail(cls, code: int, msg: str) -> "Result[None]":
        """创建统一失败响应。"""
        return cls(code=code, msg=msg, data=None)
